### PR TITLE
Fix error number is detected wrongly in tracker mode

### PR DESCRIPTION
**Pull request by tsteur**
Eg when the error is `Error query: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry`, it would detect 2300 instead of 1062.
fix DEV-1683

### CODE REVIEW

**Comment by tsteur** [comment]
Now PDO and MySQLi uses same code in tracker and regular mode. Ideally would refactor it to use the same method in the background. Not sure if I find time.

**Comment by tsteur** [comment]
Now reusing same code and added a basic test

File: core/Tracker/Db/Pdo/Mysql.php [hunk 1]
```diff
@@ -260,10 +260,7 @@ public function lastInsertId()
      */
     public function isErrNo($e, $errno)
     {
-        if (preg_match('/([0-9]{4})/', $e->getMessage(), $match)) {
-            return $match[1] == $errno;
-        }
-        return false;
+        return \Piwik\Db\Adapter\Pdo\Mysql::isPdoErrorNumber($e, $errno);
```
Comment (diosmosis) [inline core/Tracker/Db/Pdo/Mysql.php]: Does this mean the `\Piwik\Db\Adapter\Pdo\...` classes need to be required in `/piwik.php` now?

Comment (mattab) [reply core/Tracker/Db/Pdo/Mysql.php]: will merge now but @tsteur if this needs an update please go ahead and merge it in a separate PR

Comment (tsteur) [reply core/Tracker/Db/Pdo/Mysql.php]: @diosmosis not sure what you mean? The files will be auto loaded if that's what you mean?

Comment (diosmosis) [reply core/Tracker/Db/Pdo/Mysql.php]: I thought autoloading wasn't used in the tracker side for performance (hence the `require_once`s in `PIWIK_INCLUDE_PATH/piwik.php`). I guess it is used?

Comment (tsteur) [reply core/Tracker/Db/Pdo/Mysql.php]: It is definitely used as it's needed for all the plugins etc. those requires could likely be removed.

**Pull request merged**